### PR TITLE
Add query parameters for selected facet values and query params

### PR DESCRIPTION
**Pull request by wnojopra**
https://screencast.googleplex.com/cast/NTM0MTI5ODAzMDU0Mjg0OHw5NjJhNGUzZS1hYw

### CODE REVIEW

**Comment by wnojopra** [comment]
Looks like a new version of yapf came out, so unfortunately this pr includes a lot of python formatting fixes - otherwise circleci wouldn't pass.

File: api/data_explorer/__main__.py [hunk 1]
```diff
@@ -21,28 +21,24 @@
...
-    '--elasticsearch_url',
-    type=str,
-    help='Elasticsearch url, e.g. elasticsearch:9200',
-    default=os.environ.get('ELASTICSEARCH_URL'))
-parser.add_argument(
-    '--dataset_config_dir',
-    type=str,
-    help='Dataset config dir. Can be relative or absolute',
-    default=os.environ.get('DATASET_CONFIG_DIR'))
+parser.add_argument('--path_prefix',
```
Comment (melissachang) [inline api/data_explorer/__main__.py]: nit for PR name -- There's "selected facet values" and "extra facets", not sure what "selected facets" is

**Comment by wnojopra** [comment]
@melissachang yapf fixes are out of this PR, so its ready for review now

File: ui/src/App.js [hunk 2]
```diff
@@ -390,6 +395,83 @@ class App extends Component {
...
+  updateQueryParams(facetFilterParam, extraFacetsParam) {
+    window.history.pushState(
+      null,
+      "",
+      "?filter=" + facetFilterParam + "&extraFacets=" + extraFacetsParam
+    );
+  }
+
+  // Wrap the call to facetsGet together with updating the query params
+  callFacetsAPIGet(data, facetsCallback) {
```
Comment (melissachang) [inline ui/src/App.js]: nit callFacetsApiGet

That's what style guides for most languages do (eg [js](https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined)).

Also queryStringToJson

File: ui/src/App.js [hunk 3]
```diff
@@ -390,6 +395,83 @@ class App extends Component {
...
+    var pairs = window.location.search.slice(1).split("&");
+    var result = {};
+    pairs.forEach(function(pair) {
+      pair = pair.split("=");
+      result[pair[0]] = pair[1] ? decodeURIComponent(pair[1]).split("|") : [];
+    });
+    return result;
+  }
+
+  updateQueryParams(facetFilterParam, extraFacetsParam) {
```
Comment (melissachang) [inline ui/src/App.js]: nit this method is one line and only called once, can this be inlined instead?

**Pull request merged**